protocols/messages.go

```diff
@@ -91,6 +100,20 @@ func (m Message) SignedProposals() []ObjectivePayload[consensus_channel.SignedPr
 	return signedProposals
 }
 
+// SignedProposals returns a slice of signed proposals with their objectiveId that were contained in the message.
+// The proposals are sorted by ledger id then turnNum.
+func (m Message) RejectedObjectives() []ObjectivePayload[ObjectiveId] {
+	rejectedObjectives := make([]ObjectivePayload[ObjectiveId], 0)
+	for _, p := range m.payloads {
+		if p.Type() == RejectionNoticePayload {
+			entry := ObjectivePayload[ObjectiveId]{p.ObjectiveId, p.ObjectiveId}
```

Comment:
This looks a bit smelly. See my other comment about duplicating `p.ObjectiveId`.

Reply:
Resolved in 8127ead437b133b8914a5f14b5c55eb3ea3710c5